cpp/cmake_modules/Findutf8proc.cmake

```
@@ -15,6 +15,31 @@
 # specific language governing permissions and limitations
 # under the License.
 
+function(EXTRACT_UTF8PROC_VERSION)
+  if(utf8proc_INCLUDE_DIR)
+    file(READ "${utf8proc_INCLUDE_DIR}/utf8proc.h" UTF8PROC_H_CONTENT)
+
+    string(REGEX MATCH "#define UTF8PROC_VERSION_MAJOR [0-9]+" UTF8PROC_MAJOR_VERSION_DEFINITION
+            "${UTF8PROC_H_CONTENT}")
+    string(REGEX MATCH "#define UTF8PROC_VERSION_MINOR [0-9]+" UTF8PROC_MINOR_VERSION_DEFINITION
+            "${UTF8PROC_H_CONTENT}")
+    string(REGEX MATCH "#define UTF8PROC_VERSION_PATCH [0-9]+" UTF8PROC_PATCH_VERSION_DEFINITION
+            "${UTF8PROC_H_CONTENT}")
+
+    string(REGEX MATCH " [0-9.]+" UTF8PROC_MAJOR_VERSION "${UTF8PROC_MAJOR_VERSION_DEFINITION}")
+    string(REGEX MATCH " [0-9.]+" UTF8PROC_MINOR_VERSION "${UTF8PROC_MINOR_VERSION_DEFINITION}")
+    string(REGEX MATCH " [0-9.]+" UTF8PROC_PATCH_VERSION "${UTF8PROC_PATCH_VERSION_DEFINITION}")
+
+    string(STRIP "${UTF8PROC_MAJOR_VERSION}" UTF8PROC_MAJOR_VERSION)
+    string(STRIP "${UTF8PROC_MINOR_VERSION}" UTF8PROC_MINOR_VERSION)
+    string(STRIP "${UTF8PROC_PATCH_VERSION}" UTF8PROC_PATCH_VERSION)
+
+    set(UTF8PROC_VERSION "${UTF8PROC_MAJOR_VERSION}.${UTF8PROC_MINOR_VERSION}.${UTF8PROC_PATCH_VERSION}" PARENT_SCOPE)
```

Could you use `utf8proc_` prefix for variable that is exported because this package's name is `utf8proc`?

```suggestion
    set(utf8proc_VERSION "${UTF8PROC_MAJOR_VERSION}.${UTF8PROC_MINOR_VERSION}.${UTF8PROC_PATCH_VERSION}" PARENT_SCOPE)
```

---

The suggestion was applied!